src/server/routes/consents.ts

```diff
@@ -261,6 +269,226 @@ router.get(Routes.CONSENTS, loginMiddleware, (_: Request, res: Response) => {
+      NewsletterMap.get(geocode) as string[],
+      ip,
+      sc_gu_u,
+    )
+  ).slice(0, 1); // Assume 'Today' newsletter is the leading newsletter in NewsletterMap;
+  return [IDAPIEntity.NEWSLETTERS, newsletters];
+}
+
+async function getConsentEntity(
+  ip: string,
+  sc_gu_u: string,
+): Promise<[IDAPIEntity, Consent[]]> {
+  const consents = await getUserConsentsForPage(
+    [Consents.SUPPORTER],
+    ip,
+    sc_gu_u,
+  );
+  return [IDAPIEntity.CONSENTS, consents];
+}
+
+async function setNewsletterEntity(
+  ip: string,
+  sc_gu_u: string,
+  body: { [key: string]: string },
+  geocode?: GeoLocation,
+) {
+  const { update } = consentPages[1]; // Can reuse newsletter updater function.
+  if (update) {
+    await update(ip, sc_gu_u, body, geocode);
+  } else {
+    throw 'Follow On Consent Update Failure: Newsletters';
+  }
+}
+
+async function setConsentEntity(
+  ip: string,
+  sc_gu_u: string,
+  body: { [key: string]: string },
+) {
+  const consents = [
+    {
+      id: Consents.SUPPORTER,
+      consented: getConsentValueFromRequestBody(Consents.SUPPORTER, body),
+    },
+  ];
+  await patchConsents(ip, sc_gu_u, consents);
+}
+
+function getABTestGETHandler(
+  entityGetter: IDAPIEntityGetter,
+  pageTitle: string,
+) {
+  return async (req: Request, res: ResponseWithRequestState) => {
+    const { ip, cookies } = req;
+    const sc_gu_u = cookies.SC_GU_U;
+    let state = res.locals;
+    let status = 200;
+    const geocode = state.pageData.geolocation;
+    const route = req.route.path;
+    try {
+      const [entitiesName, entities] = await entityGetter(ip, sc_gu_u, geocode);
+      if (isSubscribed(entities)) {
+        res.redirect(303, getRedirectUrl(getConfiguration(), state));
+      }
+      state = {
+        ...state,
+        pageData: {
+          ...state.pageData,
+          [entitiesName]: entities,
+          returnUrl: state?.queryParams?.returnUrl,
+        },
+      };
+    } catch (error) {
+      [status, state] = getErrorResponse(error, state);
+    }
+
+    const html = renderer(route, {
+      requestState: state,
+      pageTitle,
+    });
+
+    res.type('html').status(status).send(html);
+  };
+}
+
+function getABTestPOSTHandler(
+  entitySetter: IDAPIEntitySetter,
+  pageTitle: string,
+) {
+  return async (req: Request, res: ResponseWithRequestState) => {
+    let state = res.locals;
+    const sc_gu_u = req.cookies.SC_GU_U;
+    let status = 200;
+    const url = getRedirectUrl(getConfiguration(), state);
+    const route = req.route.path;
+
+    try {
+      await entitySetter(req.ip, sc_gu_u, req.body, state.pageData.geolocation);
+      const html = redirectRenderer(url);
+      res.type('html').send(html);
+      return;
+    } catch (e) {
+      [status, state] = getErrorResponse(e, state);
+    }
+
+    const html = renderer(route, {
+      pageTitle,
+      requestState: state,
+    });
+    res
+      .type('html')
+      .status(status ?? 500)
```

Comment:
Why the `??` ?, can status be null-ish here?

Reply:
It can't be, and that's a problem ... it shouldn't default to a success! Same with the other handler, I will fix both.